pkg/rpc/server.go

```diff
@@ -121,18 +123,12 @@ func (s *Server) Run() {
 }
 
 func (s *Server) Stop(timeout time.Duration) {
-	s.logger.Info("Server is going to sleep 10 seconds before shutting down")
-	// When the sigterm signal is sent, sometimes the app could get the signal before envoy,
-	// when it does, the requests will fail because the app cannot receive any request after the shutdown.
-	// So we wait a bit in case there are still requests to be processed
-	// between the envoy and app after the signal.
-	time.Sleep(10 * time.Second)
-	s.logger.Info("Server is awakening from sleep, and going to shutdown")
+	s.logger.Info("Server is going to shutdown")
```

Comment:
Can you add more logs to check how long the shutdown process takes and if the container didn't get killed during the shutdown process?

```suggestion
	s.logger.Info("Server is going to shut down")
	ctx, cancel := context.WithTimeout(context.Background(), timeout)
	defer cancel()
	err := s.httpServer.Shutdown(ctx)
	if err != nil {
		s.logger.Error("Server failed to shut down," zap.Error(err))
	}
	s.logger.Info("Server has shut down gracefully")
```

Or we could add the elapsed time in the log after the shutdown, making it easier to read the logs.
Whatever you decide works great for me.